internal/extproc/processor.go

```diff
@@ -33,13 +26,18 @@ type processorConfig struct {
 	backendAuthHandlers                          map[string]backendauth.Handler
 	metadataNamespace                            string
 	requestCosts                                 []processorConfigRequestCost
+	declaredModels                               []string
 }
 
+// processorConfigRequestCost is the configuration for the request cost.
 type processorConfigRequestCost struct {
 	*filterapi.LLMRequestCost
 	celProg cel.Program
 }
 
+// ProcessorFactory is the factory function used to create new instances of a processor.
+type ProcessorFactory func(*processorConfig, *slog.Logger) ProcessorIface
```

Comment:
The processor config is private and in the `internal` package.
Following the idea of the `Router` in the `x` experimental API package, would it make sense (not for this PR), to move the processor interfaces there too, to allow to easily extend the processors and let users build an extproc registering their own ones?

Reply:
as per my comment above let's wait for someone to request for this being exposed for now...

Reply:
sounds good!